packages/ra-auth-auth0/src/authProvider.ts

```diff
@@ -138,15 +138,27 @@ export const Auth0AuthProvider = (
         throw new Error('Failed to get identity.');
     },
     async handleCallback() {
-        const query = window.location.search;
-        if (query.includes('code=') && query.includes('state=')) {
-            try {
-                await client.handleRedirectCallback();
-                return;
-            } catch (error) {
-                throw { redirectTo: false, message: error.message };
-            }
+        if (!handleCallbackPromise) {
+            handleCallbackPromise = new Promise(async (resolve, reject) => {
+                const query = window.location.search;
```

Comment:
Since the Promise is cached, it won't detect any change in `window.location.search`. So e.g. when accepting an email invite that expired, and trying a second invite, the user won't be able to log in.

Reply:
can't you use a debounce instead?

Reply:
It's cached as long as the `authProvider.ts` source code is in memory, but to get on the `/auth-callback` page you have to come from the Auth0 website, so there is no way the promise will still be in memory. If ever you switch user, you have to navigate to Auth0's website at some point which will unload `authProvider.ts`, so this shouldn't be an issue either.